core/bootstrap.php

```diff
@@ -15,7 +15,15 @@
     define('PIWIK_USER_PATH', PIWIK_DOCUMENT_ROOT);
 }
 
-error_reporting(E_ALL);
+$errorLevel = E_ALL;
+
+// We cannot enable deprecations for PHP 8.4 until we were able to update php-di/php-di to a version compatible
+// with PHP 8.4. Otherwise deprecation notices would be triggered at a point, where they break Matomo
```

Comment:
```suggestion
// We cannot enable deprecations for PHP 8.4 until we are able to update php-di/php-di to a version compatible
// with PHP 8.4. Otherwise deprecation notices would be triggered at a point where they break Matomo completely.
```